### PR TITLE
Separate ExecutionPlan op construction logic into different files

### PR DESCRIPTION
**Pull request by jeffreylovitz**
This is a separation of the op-building code to outside of `execution_plan.c`, no logic has been changed.

Let me know what you think and whether you'd like anything to be structured differently!

### CODE REVIEW

File: src/execution_plan/execution_plan_build/convert_clauses.c [hunk 1]
```diff
@@ -0,0 +1,243 @@
+#include "../execution_plan.h"
+#include "../../RG.h"
+#include "../ops/ops.h"
+#include "../../query_ctx.h"
+#include "../../util/rax_extensions.h"
+#include "../../ast/ast_build_ar_exp.h"
+#include "../ops/shared/scan_functions.h"
+#include "../../ast/ast_build_op_contexts.h"
```
Comment (swilly22) [inline src/execution_plan/execution_plan_build/convert_clauses.c:8]: Please make sure every include is indeed needed

File: src/execution_plan/execution_plan_build/convert_clauses.c [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+/* Reduces a filter operation into an apply operation. */
+void ExecutionPlan_ReduceFilterToApply(ExecutionPlan *plan, OpFilter *filter);
+
+/* Clause conversion logic in other files. */
+void buildMatchOpTree(ExecutionPlan *plan, AST *ast, const cypher_astnode_t *clause);
+void buildReturnOps(ExecutionPlan *plan, const cypher_astnode_t *clause);
+void buildWithOps(ExecutionPlan *plan, const cypher_astnode_t *clause);
+void buildMergeOp(ExecutionPlan *plan, AST *ast, const cypher_astnode_t *clause, GraphContext *gc);
```
Comment (swilly22) [inline src/execution_plan/execution_plan_build/convert_clauses.c:17]: I prefer moving these to a header file, including `ExecutionPlanSegment_ConvertClause`

File: src/execution_plan/execution_plan_build/build_projection_ops.c [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+		exp->resolved_name = identifier;
+
+		with_expressions = array_append(with_expressions, exp);
+	}
+
+	return with_expressions;
+
+}
+
+
```
Comment (swilly22) [inline src/execution_plan/execution_plan_build/build_projection_ops.c:134]: Remove extra line

**Review by swilly22 — CHANGES_REQUESTED** [review]
This is so much cleaner, its been overdue. just a few comments.

File: src/execution_plan/execution_plan_build/execution_plan_modify.c [hunk 4]
```diff
@@ -99,6 +99,12 @@ void ExecutionPlan_NewRoot(OpBase *old_root, OpBase *new_root) {
 	_OpBase_AddChild(tail, old_root);
 }
 
+inline void ExecutionPlan_UpdateRoot(ExecutionPlan *plan, OpBase *new_root) {
```
[the remark this answers is not in the document]
Comment (jeffreylovitz) [reply src/execution_plan/execution_plan_build/execution_plan_modify.c:102]: Yes - same logic, just migrated into this file.

File: src/execution_plan/execution_plan_build/execution_plan_modify.h [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+/* Update the root op of the execution plan. */
+void ExecutionPlan_UpdateRoot(ExecutionPlan *plan, OpBase *new_root);
+
+/* Replace a with b. */
+void ExecutionPlan_ReplaceOp(ExecutionPlan *plan, OpBase *a, OpBase *b);
+
+/*
+ * ExecutionPlan_Locate API:
+ * For performing existence checks and looking up individual operations in tree.
+ */
```
Comment (swilly22) [inline src/execution_plan/execution_plan_build/execution_plan_modify.h:43]: Consider using GraphBLAS comment "notation"
```
//-----------------------------
// DESC
//-----------------------------
```

**Review by swilly22 — APPROVED** [review]
Please see comments.

**Review by swilly22 — APPROVED** [review]

**Pull request merged**